lib/fog/openstack/core.rb

```diff
@@ -78,13 +78,14 @@ def reload
       def request(params, parse_json = true)
         retried = false
         begin
+          _path=@path
```

Comment:
ruby doesn't usually use the python _ and __ prefixes. Just put there 'path'. _ in ruby marks it's not used, e.g. unused param.

Might be that the error is actually in the authenticate? What is being placed in the path, when authenticate is called? It should be always set to the right value.